tor/net.go

```diff
@@ -42,13 +50,45 @@ type Net interface {
 
 // ClearNet is an implementation of the Net interface that defines behaviour
 // for regular network connections.
-type ClearNet struct{}
+type ClearNet struct {
+	// SOCKS is the host:port for SOCKS5 proxy to use for clearnet connections.
+	SOCKS string
+
+	// SOCKSAuth contains credentials to use for authentication to SOCKS proxy.
+	SOCKSAuth *proxy.Auth
+
+	// NoProxyTargets is a string of comma-separated values
+	// specifying hosts that should bypass the proxy. Each value is either an
+	// IP address, a CIDR range, a zone (*.example.com) or a host name
+	// (localhost). A best effort is made to parse the string and errors are
+	// ignored.
+	NoProxyTargets string
+}
+
+func (r *ClearNet) createDialer(timeout time.Duration) (Dialer, error) {
+	clearDialer := &net.Dialer{Timeout: timeout}
+	if r.SOCKS == "" {
+		return clearDialer, nil
+	}
+
+	dialer, err := proxy.SOCKS5("tcp", r.SOCKS, r.SOCKSAuth, clearDialer)
+	if err != nil {
+		return nil, err
+	}
+	perHostDialer := proxy.NewPerHost(dialer, clearDialer)
```

Comment:
nit: add empty line after an `if err != nil {` block to add some vertical spacing.